[Discover] Improve sidebar rendering for large field lists

kertal opened the pull request:
## Summary

This PR significantly improves rendering speed of Discover's sidebar field selector.  This is done by implementing  pagination while scrolling the list of available fields. Initially only 50 fields are rendered, while you are scrolling the sidebar list, fields in chunks of 50 are added and rendered. Code inspired by Lens.

### Testing

You can use the following dataset which has thousands of fields  https://gist.github.com/flash1293/b98acceef7212019ee8349bea16fb1a5 for testing. You need to create an index pattern for it for testing.

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/)
* Commit: 0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c0f9bfcd217f20a71818e432e470cbd01875349d) | [after](https://github.com/elastic/kibana/commit/0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4) | diff |
| --- | --- | --- | --- |
| `discover` | 383.6KB | 384.4KB | +892.0B |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/","number":"114255","commit":"0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4"}]}
PIPELINE-->

flash1293 commented:
I tested this PR with https://gist.github.com/flash1293/b98acceef7212019ee8349bea16fb1a5 and it's super fast!

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/)
* Commit: f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/execution/node/612/log)
* [Execute kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/execution/node/788/log)

### History
* :green_heart: [Build #114255](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/) succeeded 0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/","number":"114255","commit":"0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/","number":"114624","commit":"f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/)
* Commit: bcaac9200887567d96065d6c2a6bcd16bd49e7fe
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/bcaac9200887567d96065d6c2a6bcd16bd49e7fe)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/execution/node/612/log)
* [Execute kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/execution/node/794/log)

### History
* :broken_heart: [Build #114624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/) failed f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2
* :green_heart: [Build #114255](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/) succeeded 0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/","number":"114255","commit":"0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/","number":"114624","commit":"f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/","number":"114926","commit":"bcaac9200887567d96065d6c2a6bcd16bd49e7fe"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/)
* Commit: de50c93ca9859410b17723ff930db60309a202e5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/de50c93ca9859410b17723ff930db60309a202e5)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/cb61b4a8fd07069396c9697a1d0d386dcf65e75f) | [after](https://github.com/elastic/kibana/commit/de50c93ca9859410b17723ff930db60309a202e5) | diff |
| --- | --- | --- | --- |
| `discover` | 388.0KB | 389.2KB | +1.3KB |

### History
* :broken_heart: [Build #114926](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/) failed bcaac9200887567d96065d6c2a6bcd16bd49e7fe
* :broken_heart: [Build #114624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/) failed f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2
* :green_heart: [Build #114255](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/) succeeded 0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/","number":"114255","commit":"0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/","number":"114624","commit":"f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/","number":"114926","commit":"bcaac9200887567d96065d6c2a6bcd16bd49e7fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/","number":"115007","commit":"de50c93ca9859410b17723ff930db60309a202e5"}]}
PIPELINE-->

flash1293 reviewed (commented):
Tested and this makes the side bar much faster. The document column in the table is still a huge issue, but I think that will be addressed separately.

One thing that might be in-scope for this PR is this field search issue: It's very quick filtering fields, but on trying to remove the filter, it just freezes:
![freezing](https://user-images.githubusercontent.com/1508364/112475391-1189da80-8d71-11eb-9a31-dc1e92379905.gif)

In the end of the gif, I hit backspace twice to go back to `field4`, but this is so slow it triggers the "page is not responding" dialog

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/)
* Commit: ab49c6f48f46ab6a55035990f3fc56efe274db0d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/ab49c6f48f46ab6a55035990f3fc56efe274db0d)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6b5023a681673a0930b2dd34c203bbca3cb0beac) | [after](https://github.com/elastic/kibana/commit/ab49c6f48f46ab6a55035990f3fc56efe274db0d) | diff |
| --- | --- | --- | --- |
| `discover` | 388.9KB | 390.2KB | +1.3KB |

### History
* :green_heart: [Build #115007](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/) succeeded de50c93ca9859410b17723ff930db60309a202e5
* :broken_heart: [Build #114926](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/) failed bcaac9200887567d96065d6c2a6bcd16bd49e7fe
* :broken_heart: [Build #114624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/) failed f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2
* :green_heart: [Build #114255](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/) succeeded 0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/","number":"114255","commit":"0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/","number":"114624","commit":"f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/","number":"114926","commit":"bcaac9200887567d96065d6c2a6bcd16bd49e7fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/","number":"115007","commit":"de50c93ca9859410b17723ff930db60309a202e5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/","number":"115148","commit":"ab49c6f48f46ab6a55035990f3fc56efe274db0d"}]}
PIPELINE-->

kertal commented on src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx:
```diff
@@ -112,6 +124,34 @@ export function DiscoverSidebar({
+      if (clientHeight === scrollHeight && pageSize < unpopularFields.length) {
+        // in case of there is so much height available that there's no scrolling at initial rendering
+        // pagination starts after the first bulk of rendering
+        setTimeout(() => paginate(), 100);
```
@flash1293 was't completely done here, so I've adapted the code that caused your troubles. But I'm not certain that this could not cause other issues. 

I tried to fix the following case: there are 50 fields of 3000 rendered, but due to zooming or a very large vertical monitor, there is no scrolling necessary for those 50 rendered fields. Then you this kind of pagination wouldn't work, you would stick to 50 fields. any ideas, or just set to 100?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115220/)
* Commit: 06566d6d63cb33dd9da7270045537b20f77c0e2d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/06566d6d63cb33dd9da7270045537b20f77c0e2d)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6b5023a681673a0930b2dd34c203bbca3cb0beac) | [after](https://github.com/elastic/kibana/commit/06566d6d63cb33dd9da7270045537b20f77c0e2d) | diff |
| --- | --- | --- | --- |
| `discover` | 388.9KB | 390.2KB | +1.3KB |

### History
* :green_heart: [Build #115148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/) succeeded ab49c6f48f46ab6a55035990f3fc56efe274db0d
* :green_heart: [Build #115007](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/) succeeded de50c93ca9859410b17723ff930db60309a202e5
* :broken_heart: [Build #114926](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/) failed bcaac9200887567d96065d6c2a6bcd16bd49e7fe
* :broken_heart: [Build #114624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/) failed f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2
* :green_heart: [Build #114255](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/) succeeded 0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114255/","number":"114255","commit":"0bd248e8d6ddbf23a0f67b793cea317bab1b3ca4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/","number":"114624","commit":"f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/","number":"114926","commit":"bcaac9200887567d96065d6c2a6bcd16bd49e7fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/","number":"115007","commit":"de50c93ca9859410b17723ff930db60309a202e5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/","number":"115148","commit":"ab49c6f48f46ab6a55035990f3fc56efe274db0d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115220/","number":"115220","commit":"06566d6d63cb33dd9da7270045537b20f77c0e2d"}]}
PIPELINE-->

flash1293 commented:
@kertal I took me a while to understand what's going on 😁

So I get the problem correctly - if the initial render of items is too low for the screen, it will never start scrolling because it will never enter overflow and thus won't cause scroll events.

The cleanest approach I can think of is to use a resize observer to wait for the list container to update its size, then do a check whether it doesn't fill its parent (if that's the case, load more items). It shouldn't suffer from the same problem as you r current code - if I understand correctly react bumps the size multiple times because the DOM doesn't get a change to re-layout and update its height.

That being said, using the viewport height to estimate the number of initial list items, then adding a buffer to it seems like an easier approach - something like this:
```ts
// initially load at least 50 items, but increase the number if on tall screens
useState(Math.max(PAGINATION_SIZE, Math.round(window.innerHeight / 25 + 20)));
```

. Technically, Lens has the same issue, but I don't know of any reports (granted, you have to zoom out a lot):
<img width="749" alt="Screenshot 2021-03-27 at 15 05 52" src="https://user-images.githubusercontent.com/1508364/112723317-1768f280-8f0e-11eb-85fa-3549b3a318c7.png">

kertal commented:
> @kertal I took me a while to understand what's going on 😁

@flash1293 also took me a while to discover this edgy case :)
> 
> Just to make sure I get the problem correctly - if the initial render of items is too low for the screen, it will never start scrolling because it will never enter overflow and thus won't cause scroll events.

exactly

> 
> The cleanest approach I can think of is to use a resize observer to wait for the list container to update its size, then do a check whether it doesn't fill its parent (if that's the case, load more items). It shouldn't suffer from the same problem as you r current code - if I understand correctly react bumps the size multiple times because the DOM doesn't get a change to re-layout and update its height.
> 
> That being said, using the viewport height to estimate the number of initial list items, then adding a buffer to it seems like an easier approach - something like this:
> 
> ```ts
> // initially load at least 50 items, but increase the number if on tall screens
> useState(Math.max(PAGINATION_SIZE, Math.round(window.innerHeight / 25 + 20)));
> ```

thx, so I've implemented a similar version, don't use window.innerHeight, calculate the average height of a field (they can be of different size) to see how much would fit in the container, and assign a custom size of fields to render in this case,

> 
> It could still break if the user starts with a low-height screen, then resizes it (or zooms out), so the height grows until the element isn't scrollable anymore. But that's pretty theoretical.

I've added `useResizeObserver` for the container and this works 

> 
> Technically, Lens has the same issue, but I don't know of any reports (granted, you have to zoom out a lot):
> <img alt="Screenshot 2021-03-27 at 15 05 52" width="749" src="https://user-images.githubusercontent.com/1508364/112723317-1768f280-8f0e-11eb-85fa-3549b3a318c7.png">

it is for sure a minority issue, for people with rotated 50 inch monitors ... and those who google earth like zoom out settings

flash1293 commented:
> thx, so I've implemented a similar version, don't use window.innerHeight, calculate the average height of a field (they can be of different size) to see how much would fit in the container, and assign a custom size of fields to render in this case,

Even better! Played around with it, works like a charm now.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115750/)
* Commit: 4769c9854ec4fc425429724adb8b3c389f4ad483
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/4769c9854ec4fc425429724adb8b3c389f4ad483)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115750/execution/node/722/log)
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115750/execution/node/851/log)

### History
* :green_heart: [Build #115220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115220/) succeeded 06566d6d63cb33dd9da7270045537b20f77c0e2d
* :green_heart: [Build #115148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/) succeeded ab49c6f48f46ab6a55035990f3fc56efe274db0d
* :green_heart: [Build #115007](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/) succeeded de50c93ca9859410b17723ff930db60309a202e5
* :broken_heart: [Build #114926](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/) failed bcaac9200887567d96065d6c2a6bcd16bd49e7fe
* :broken_heart: [Build #114624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/) failed f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114624/","number":"114624","commit":"f3a0e4ce4e19e72fb94cb0ff5f9286ef8d7fc1f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/","number":"114926","commit":"bcaac9200887567d96065d6c2a6bcd16bd49e7fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/","number":"115007","commit":"de50c93ca9859410b17723ff930db60309a202e5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/","number":"115148","commit":"ab49c6f48f46ab6a55035990f3fc56efe274db0d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115220/","number":"115220","commit":"06566d6d63cb33dd9da7270045537b20f77c0e2d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115750/","number":"115750","commit":"4769c9854ec4fc425429724adb8b3c389f4ad483"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115770/)
* Commit: 30b4e0bd8bebb5f05446ec6858f02af24814084c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/30b4e0bd8bebb5f05446ec6858f02af24814084c)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115770/execution/node/708/log)
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115770/execution/node/908/log)

### History
* :broken_heart: [Build #115750](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115750/) failed 4769c9854ec4fc425429724adb8b3c389f4ad483
* :green_heart: [Build #115220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115220/) succeeded 06566d6d63cb33dd9da7270045537b20f77c0e2d
* :green_heart: [Build #115148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/) succeeded ab49c6f48f46ab6a55035990f3fc56efe274db0d
* :green_heart: [Build #115007](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/) succeeded de50c93ca9859410b17723ff930db60309a202e5
* :broken_heart: [Build #114926](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/) failed bcaac9200887567d96065d6c2a6bcd16bd49e7fe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114926/","number":"114926","commit":"bcaac9200887567d96065d6c2a6bcd16bd49e7fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/","number":"115007","commit":"de50c93ca9859410b17723ff930db60309a202e5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/","number":"115148","commit":"ab49c6f48f46ab6a55035990f3fc56efe274db0d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115220/","number":"115220","commit":"06566d6d63cb33dd9da7270045537b20f77c0e2d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115750/","number":"115750","commit":"4769c9854ec4fc425429724adb8b3c389f4ad483"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115770/","number":"115770","commit":"30b4e0bd8bebb5f05446ec6858f02af24814084c"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115785/)
* Commit: 6a55a5a75ab60d8a0803a1a358a101fc0859f5a4
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95038/6a55a5a75ab60d8a0803a1a358a101fc0859f5a4)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d16101f377bc67dae4900701b4ffe4604247683c) | [after](https://github.com/elastic/kibana/commit/6a55a5a75ab60d8a0803a1a358a101fc0859f5a4) | diff |
| --- | --- | --- | --- |
| `discover` | 388.9KB | 391.1KB | +2.2KB |

### History
* :broken_heart: [Build #115770](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115770/) failed 30b4e0bd8bebb5f05446ec6858f02af24814084c
* :broken_heart: [Build #115750](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115750/) failed 4769c9854ec4fc425429724adb8b3c389f4ad483
* :green_heart: [Build #115220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115220/) succeeded 06566d6d63cb33dd9da7270045537b20f77c0e2d
* :green_heart: [Build #115148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/) succeeded ab49c6f48f46ab6a55035990f3fc56efe274db0d
* :green_heart: [Build #115007](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/) succeeded de50c93ca9859410b17723ff930db60309a202e5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kertal

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115007/","number":"115007","commit":"de50c93ca9859410b17723ff930db60309a202e5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115148/","number":"115148","commit":"ab49c6f48f46ab6a55035990f3fc56efe274db0d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115220/","number":"115220","commit":"06566d6d63cb33dd9da7270045537b20f77c0e2d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115750/","number":"115750","commit":"4769c9854ec4fc425429724adb8b3c389f4ad483"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115770/","number":"115770","commit":"30b4e0bd8bebb5f05446ec6858f02af24814084c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115785/","number":"115785","commit":"6a55a5a75ab60d8a0803a1a358a101fc0859f5a4"}]}
PIPELINE-->

flash1293 reviewed (approved):
Tested in Chrome, Firefox and Safari and it works fine. I think there's still potential for further optimization, but this PR is the difference between "not usable at all" and works quite well for large documents. Great job!
